### PR TITLE
Fix aspect ratio sampling for RandomResizedCrop

### LINKED ISSUE #14505: RandomResizedCrop produces wrong aspect ratios.
RandomResizedCrop randomly swaps with/height which produces out-of-bounds aspect ratios if the ends of the aspect ratio range are not reciprocals. When swapping is removed, then the aspect ratio is skewed. The solution is to use uniform distribution on a logarithmic scale. See https://github.com/pytorch/vision/pull/799

### PR DESCRIPTION
**Pull request by abhinavs95**
## Description ##
Fixes #14505 

Aspect ratio goes out of bounds when the bounds are not reciprocals. Fixed by sampling from log scale instead of linear scale. When sampling from log scale, random swapping of width/height is not required since the distribution is centered at 1.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by abhinavs95** [comment]
@zhreshold @szha Could you have a look?

@mxnet-label-bot add [pr-awaiting-review]

**Comment by marcoabreu** [comment]
Thanks for your contribution, would you mind adding a test case for this situation?

**Comment by abhinavs95** [comment]
@marcoabreu added test in test_image.py

**Comment by piyushghai** [comment]
@abhinavs95 Can you look into the CI-failures ?

File: python/mxnet/image/image.py [hunk 1]
```diff
@@ -585,14 +586,12 @@ def random_size_crop(src, size, area, ratio, interp=2, **kwargs):
         area = (area, 1.0)
     for _ in range(10):
         target_area = random.uniform(area[0], area[1]) * src_area
-        new_ratio = random.uniform(*ratio)
+        log_ratio = (math.log(ratio[0]), math.log(ratio[1]))
```
Comment (zhreshold) [inline python/mxnet/image/image.py]: you don't need to import `math`, use `np` which is already imported

Comment (abhinavs95) [reply python/mxnet/image/image.py]: done!

**Comment by abhinavs95** [comment]
@szha could you have a look and merge? Thanks

@mxnet-label-bot update [pr-awaiting-merge]

**Pull request merged**